src/Kestrel.Core/Internal/Http2/Http2Connection.cs

```diff
@@ -89,6 +89,8 @@ public Http2Connection(Http2ConnectionContext context)
             _frameWriter = new Http2FrameWriter(context.Transport.Output, context.ConnectionContext, _outputFlowControl, this, context.ConnectionId, context.ServiceContext.Log);
             _hpackDecoder = new HPackDecoder((int)_serverSettings.HeaderTableSize);
             _serverSettings.MaxConcurrentStreams = (uint)context.ServiceContext.ServerOptions.Limits.Http2.MaxStreamsPerConnection;
+            _serverSettings.MaxFrameSize = (uint)context.ServiceContext.ServerOptions.Limits.Http2.MaxFrameSize;
+            _incomingFrame = new Http2Frame(_serverSettings.MaxFrameSize);
```

Comment:
This allocates a 16 MB buffer heap allocated buffer per connection... ?

Reply:
Yea... we haven't really optimized any of this and have been allocating all over the place.

Reply:
Should we do this now? I think there are tons of allocations everywhere and we should do it systematically at some point. But I suppose we could start piecemeal at a time.